movement/analysis/kinematics.py

```diff
@@ -165,3 +169,472 @@ def compute_time_derivative(data: xr.DataArray, order: int) -> xr.DataArray:
+
+    """
+    return _compute_pairwise_distances(
+        data, "keypoints", pairs=pairs, metric=metric
+    )
+
+
+def _compute_pairwise_distances(
+    data: xr.DataArray,
+    dim: Literal["individuals", "keypoints"],
+    pairs: dict[str, str | list[str]] | None = None,
+    metric: str | None = "euclidean",
+    **kwargs,
+) -> xr.DataArray | dict[str, xr.DataArray]:
+    """Compute pairwise distances between ``individuals`` or ``keypoints``.
+
+    This function computes the distances between pairs of ``keypoints``
+    (i.e. interkeypoint distances) or pairs of ``individuals`` (i.e.
+    interindividual distances). The distances are computed using the
+    specified ``metric``.
+
+    Parameters
+    ----------
+    data : xarray.DataArray
+        The input data containing
+        ``('time', 'individuals', 'keypoints', 'space')`` as dimensions.
+    dim : str
+        The dimension to compute the distances for. Must be either
+        ``'individuals'`` or ``'keypoints'``.
+    pairs : dict[str, str | list[str]], optional
+        A dictionary containing the mapping between pairs of ``individuals``
+        or ``keypoints``. The key is the keypoint or individual and the
+        value is a list of keypoints or individuals or a string
+        representing a keypoint or individual to compute the distance with.
+        If not provided, defaults to ``None`` and all possible combinations
+        of pairs are computed.
+    metric : str, optional
+        The distance metric to use. Must be one of the options supported
+        by :func:`scipy.spatial.distance.cdist`, i.e.
+        ``'braycurtis'``, ``'canberra'``, ``'chebyshev'``, ``'cityblock'``,
+        ``'correlation'``, ``'cosine'``, ``'dice'``, ``'euclidean'``,
+        ``'hamming'``, ``'jaccard'``, ``'jensenshannon'``, ``'kulczynski1'``,
+        ``'mahalanobis'``, ``'matching'``, ``'minkowski'``,
+        ``'rogerstanimoto'``, ``'russellrao'``, ``'seuclidean'``,
+        ``'sokalmichener'``, ``'sokalsneath'``, ``'sqeuclidean'``, ``'yule'``.
+        Defaults to ``'euclidean'``.
+    **kwargs : dict
+        Additional keyword arguments to pass to
+        :func:`scipy.spatial.distance.cdist`.
+
+    Returns
+    -------
+    xarray.DataArray | dict[str, xarray.DataArray]
+        An :class:`xarray.DataArray` containing the computed distances between
+        a single pair of ``individuals`` or ``keypoints``. If multiple
+        ``pairs`` are provided, this will be a dictionary containing the
+        computed distances for each pair, with the key being the pair of
+        keypoints or individuals and the value being the
+        :class:`xarray.DataArray` containing the computed distances.
+
+    See Also
+    --------
+    cdist : ``movement``'s wrapper around :func:`scipy.spatial.distance.cdist`.
+    :func:`scipy.spatial.distance.cdist` : The underlying function used.
+
+    """
+    if dim not in ["individuals", "keypoints"]:
```

Comment:
I think there is an extra validation needed here.

When running `compute_interindividual_distances` on a data array with only 1 individual, or `compute_interkeypoint_distances` on a data array with only 1 keypoint, the output in a empty dictionary (without any error or warning being raised). Is this the behaviour we want?

Reply:
This will now raise a ValueError